### PR TITLE
Add script support for CentOS 8

### PR DESCRIPTION
**Pull request by softprofe**
## Change Description

Start 9361 description ......
In script eos/scripts/eosio_build.sh, line 122, set -i  pop error on centos7 and conetos8, so, add OS judge and not use set -i on CentOS, tested can be built without "set -i";
In script eos/scripts/eosio_build_centos.sh, it regard centos version as 7 without judge, and install package which only can be used on centos 7. I add judge and install package which can be used on centos8.  centos 7 will still install the original packages.
End 9361 description...

## Change Type
**Select ONE**
- [ ] Documentation

- [ ] Stability bug fix

- [x] Other

- [ ] Other - special case

## Consensus Changes
- [ ] Consensus Changes

## API Changes
- [ ] API Changes

## Documentation Additions
- [ ] Documentation Additions

### CODE REVIEW

File: scripts/eosio_build_centos.sh [hunk 1]
```diff
@@ -23,6 +24,14 @@ if [[ -d /opt/rh/devtoolset-8 ]]; then
 fi
 # Ensure packages exist
 ensure-yum-packages "${REPO_ROOT}/scripts/eosio_build_centos7_deps"
+fi
+
+if [[ "$(echo ${VERSION} | sed 's/ .*//g')" == 8 ]]; then
+        echo "Install Development Tools ..."
+        dnf install -y  https://dl.fedoraproject.org/pub/epel/epel-release-latest-8.noarch.rpm
+        dnf group install -y  "Development Tools"
```
Comment (spoonincode) [inline scripts/eosio_build_centos.sh:32]: as is, these will require the script to run as root

**Comment by spoonincode** [comment]
will not work once #9198 is merged because it is lacking installation of llvm

**Comment by softprofe** [comment]
> will not work on develop once #9198 is merged because it is lacking installation of llvm
> (won't work as is on 2.0)

Based on my experience of build eos on Centos,  after dnf group install -y  "Development Tools", llvm will  still  be installed by ensure-llvm()

**Comment by spoonincode** [comment]
> llvm will still be installed by ensure-llvm()

Even for an unpinned build?

**Comment by spoonincode** [comment]
My centos8 container was populated with a "minimal" install and interestingly failed to build because the lack of openssl dev being installed. Maybe we should add that as a installed package

**Comment by softprofe** [comment]
> My centos8 container was populated with a "minimal" install and interestingly failed to build because the lack of openssl dev being installed. Maybe we should add that as a installed package

Sure we can install openssl-dev   for centos 8

**Comment by softprofe** [comment]
> > llvm will still be installed by ensure-llvm()
> 
> Even for an unpinned build?

should not, because unpinned build don't need it I think.

**Comment by spoonincode** [comment]
>should not, because unpinned build don't need it I think.

If you merge develop in to this branch I think you'll find you need it

**Comment by softprofe** [comment]
> > should not, because unpinned build don't need it I think.
> 
> If you merge develop in to this branch I think you'll find you need it
Why  ensure-llvm() won't work?

**Pull request closed without merging**